dataclients/kubernetes/ingressdefinitions.go

```diff
@@ -65,6 +65,145 @@ func (s service) getTargetPortByValue(p int) (*definitions.BackendPort, bool) {
 	return nil, false
 }
 
+// There are [1..N] Kubernetes endpointslices created for a single Kubernetes service.
+// Kubernetes endpointslices of a given service can have duplicates with different states.
+// Therefore Kubernetes endpointslices need to be de-duplicated before usage.
+// The business object skipperEndpointSlice is a de-duplicated endpoint list that concats all endpointslices of a given service into one slice of skipperEndpointSlice.
+type skipperEndpointSlice struct {
+	Meta      *definitions.Metadata
+	Endpoints []*skipperEndpoint
+	Ports     []*endpointSlicePort
+}
+
+// Conditions have to be evaluated before creation
+type skipperEndpoint struct {
+	Address string
+	Zone    string
+}
+
+func (eps skipperEndpointSlice) getPort(protocol, pName string, pValue int) int {
+	var port int
+
+	for _, p := range eps.Ports {
+		if p.Protocol != protocol {
+			continue
+		}
+		if p.Name == pName {
+			port = p.Port
+			break
+		}
+		if p.Port == pValue {
+			port = pValue
+			break
+		}
+	}
+
+	return port
+}
+func (eps skipperEndpointSlice) targetsByServicePort(protocol, backendProtocol string, servicePort *servicePort) []string {
+	port := eps.getPort(protocol, servicePort.Name, servicePort.Port)
+
+	var result []string
+	for _, ep := range eps.Endpoints {
+		result = append(result, formatEndpointString(ep.Address, backendProtocol, port))
+	}
+
+	return result
+}
+
+func (eps skipperEndpointSlice) targetsByServiceTarget(protocol, backendProtocol string, serviceTarget *definitions.BackendPort) []string {
+	pName := serviceTarget.Value.(string)
+	pValue := serviceTarget.Value.(int)
+	port := eps.getPort(protocol, pName, pValue)
+
+	var result []string
+	for _, ep := range eps.Endpoints {
+		result = append(result, formatEndpointString(ep.Address, backendProtocol, port))
+	}
+
+	return result
+}
+
+func (eps skipperEndpointSlice) targets(protocol, backendProtocol string) []string {
+	result := make([]string, 0)
+
+	var port int
+	for _, p := range eps.Ports {
+		if p.Protocol == protocol {
+			port = p.Port
+			break
+		}
+	}
+	for _, ep := range eps.Endpoints {
+		result = append(result, formatEndpointString(ep.Address, backendProtocol, port))
+	}
+
+	return result
+}
+
+type endpointSliceList struct {
+	Meta  *definitions.ListMetadata
```

Comment:
Should it be tagged as "`json:",inline"`"?

Maybe we can just use existing metadata here and simply add continue and remaining fields without introducing ListMetadata type?

Reply:
I see now that continue field is part of metadata. I think it should be tagged with "`json:"metadata"`"

Reply:
We don't need "continue". It was a miss understanding by myself but I clarified with sig-network.
The "continue" only exists in Lists because it is about number of objects in lists (for example GET all pods and if this listing is more than 1000 objects). One endpointslice holds a list of endpoints but it not a list request but a GET of a single object.

Reply:
this will addressed in the next push